app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/HomeScreenUITest.kt

```diff
@@ -0,0 +1,185 @@
+            onNodeWithTag(testTag = TestTagCardTotalCoins).assertTextEquals("$7,273,291")
+            onNodeWithTag(testTag = TestTagCardTodayProfit).assertTextEquals("$193,280")
+        }
+    }
+
+    @Test
+    fun when_enter_to_HomeScreen_and_load_MyCoins_successfully_it_render_the_UI_properly() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            with(MockUtil.myCoins.first()) {
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemSymbol,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(symbol.uppercase())
+
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemCoinName,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(coinName)
+
+                onAllNodesWithTag(
+                    testTag = TestTagCoinItemPriceChange,
+                    useUnmergedTree = true
+                ).onFirst().onChild().assertTextEquals(expectedPriceChange)
+            }
+        }
+    }
+
+    @Test
+    fun when_enter_to_HomeScreen_and_load_TrendingCoins_successfully_it_render_the_UI_properly() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            with(MockUtil.trendingCoins.first()) {
+                onAllNodesWithTag(
+                    testTag = TestTagTrendingItemSymbol,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(symbol.uppercase())
+
+                onAllNodesWithTag(
+                    testTag = TestTagTrendingItemCoinName,
+                    useUnmergedTree = true
+                ).onFirst().assertTextEquals(coinName)
+
+                onAllNodesWithTag(
+                    testTag = TestTagTrendingItemPriceChange,
+                    useUnmergedTree = true
+                ).onFirst().onChild().assertTextEquals(expectedPriceChange)
+            }
+        }
+    }
+
+    @Test
+    fun when_clicked_on_MyCoin_item_it_navigates_to_DetailScreen() {
+        initViewModel()
+
+        composeAndroidTestRule.onAllNodesWithTag(testTag = TestTagCoinItem).onFirst().performClick()
+
+        appDestination shouldBe AppDestination.CoinDetail
+    }
+
+    @Test
+    fun when_clicked_on_TrendingCoin_item_it_navigates_to_DetailScreen() {
+        initViewModel()
+
+        composeAndroidTestRule.onAllNodesWithTag(
+            testTag = TestTagTrendingItem
+        ).onFirst().performClick()
+
+        appDestination shouldBe AppDestination.CoinDetail
+    }
+
+    @Test
+    fun when_enter_to_HomeScreen_and_load_MyCoins_failed_it_shows_the_Toast_properly() {
+        every { mockGetMyCoinsUseCase.execute(any()) } returns flow {
+            throw Throwable(errorGeneric)
+        }
+
+        initViewModel()
+
+        composeAndroidTestRule.onNodeWithTag(
+            testTag = TestTagCoinItem,
+            useUnmergedTree = true
+        ).assertDoesNotExist()
+    }
```

Comment:
I don't see the code to test the Toast 🤔

Reply:
@luongvo We can see it while running the test case UI in the device 🤔 , So I don't think we need to assert anything for this here:


https://user-images.githubusercontent.com/28002315/219591719-fb88dbab-ba63-49a7-a534-33aaa38d2174.mov

Reply:
@Wadeewee we're not writing a UI test to see it for verifying lol. We need to write test code to assert what we need to expect on the screen 🙏